src/main/java/io/vertx/redis/impl/AbstractRedisService.java

```diff
@@ -0,0 +1,245 @@
+package io.vertx.redis.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.impl.LoggerFactory;
+import io.vertx.redis.RedisService;
+
+import java.nio.charset.Charset;
+
+public abstract class AbstractRedisService implements RedisService {
+
+  private static final Logger log = LoggerFactory.getLogger(AbstractRedisService.class);
+
+  private enum ResponseTransform {
+    NONE,
+    ARRAY_TO_OBJECT,
+    INFO
+  }
+
+  final Vertx vertx;
+  final JsonObject config;
+  final EventBus eb;
+
+  private RedisConnection redisClient;
+  private RedisSubscriptions subscriptions = new RedisSubscriptions();
+
+  private String encoding;
+  private Charset charset;
+  private String baseAddress;
+
+  AbstractRedisService(Vertx vertx, JsonObject config) {
+    this.vertx = vertx;
+    this.config = config;
+    this.eb = vertx.eventBus();
+  }
+
+  @Override
+  public void start(final Handler<AsyncResult<Void>> handler) {
+    final String host = config.getString("host", "localhost");
+    final int port = config.getInteger("port", 6379);
+    final String enc = config.getString("encoding", "UTF-8");
+    final boolean binary = config.getBoolean("binary", false);
+
+    if (binary) {
+      log.warn("Binary mode is not implemented yet!!!");
```

Comment:
Probably not as the service is still usable, the binary protocol just isn't supported.It's not as if they have misconfigured a host name or port or something requiring a complete failure.